display/pfobjects.py

```diff
@@ -34,11 +29,23 @@ def __init__(self, cluster):
         inners = [self.inner_xy, self.inner_yz, self.inner_xz,
                   self.inner_thetaphi]
         for contour in contours:
-            contour.SetLineColor(color)
-            contour.SetFillStyle(0)
-        for inner in inners: 
-            inner.SetFillColor(color)
-            inner.SetFillStyle(3002)
+           if cluster.particle:
+            if cluster.particle.pdgid() == 22:	
+                contour.SetLineColor(2)	
+            elif cluster.particle.pdgid() == 11:	
+                contour.SetLineColor(46)
+            else:					
+                contour.SetLineColor(4)
+            contour.SetFillStyle(0) 
+        for inner in inners:
+           if cluster.particle: 
+            if cluster.particle.pdgid() == 22:	
+                inner.SetFillColor(2)	
+            elif cluster.particle.pdgid() == 11:	
+                inner.SetFillColor(46)
```

Comment:
hardcoded numbers should not be repeated. this will be solved if you define color as done before, and before the loop. then just use color as was done before